src/OpenTelemetry/Trace/TracerProviderSdk.cs

```diff
@@ -34,6 +34,8 @@ internal sealed class TracerProviderSdk : TracerProvider
         private readonly Sampler sampler;
         private readonly Action<Activity> getRequestedDataAction;
         private readonly bool supportLegacyActivity;
+        private readonly bool legacyActivityWildcardMode;
```

Comment:
I guess making this local and capture it in lambda/delegate would be faster than accessing the value via this pointer.
I haven't tried it myself, a [benchmark](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/test/Benchmarks/Trace/TraceBenchmarks.cs) or [stress test](https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/test/OpenTelemetry.Tests.Stress) should give us the numbers.